openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,218 @@ def maintained_selection():
+    asset_data = instance.data["assetEntity"]["data"]
+    asset_width = asset_data.get("resolutionWidth")
+    asset_height = asset_data.get("resolutionHeight")
+    review_instance_width = instance.data.get("review_width")
+    review_instance_height = instance.data.get("review_height")
+    preset["camera"] = camera
+
+    # Tests if project resolution is set,
+    # if it is a value other than zero, that value is
+    # used, if not then the asset resolution is
+    # used
+    if review_instance_width and review_instance_height:
+        preset["width"] = review_instance_width
+        preset["height"] = review_instance_height
+    elif width_preset and height_preset:
+        preset["width"] = width_preset
+        preset["height"] = height_preset
+    elif asset_width and asset_height:
+        preset["width"] = asset_width
+        preset["height"] = asset_height
+    preset["start_frame"] = start
+    preset["end_frame"] = end
+
+    # Enforce persisting camera depth of field
+    camera_options = preset.setdefault("camera_options", {})
+    camera_options["depthOfField"] = cmds.getAttr(
+        "{0}.depthOfField".format(camera))
+
+    preset["filename"] = path
+    preset["overwrite"] = True
+
+    cmds.refresh(force=True)
+
+    refreshFrameInt = int(cmds.playbackOptions(q=True, minTime=True))
+    cmds.currentTime(refreshFrameInt - 1, edit=True)
+    cmds.currentTime(refreshFrameInt, edit=True)
+
+    # Use displayLights setting from instance
+    key = "displayLights"
+    preset["viewport_options"][key] = instance.data[key]
+
+    # Override transparency if requested.
+    transparency = instance.data.get("transparency", 0)
+    if transparency != 0:
+        preset["viewport2_options"]["transparencyAlgorithm"] = transparency
+
+    # Isolate view is requested by having objects in the set besides a
+    # camera. If there is only 1 member it'll be the camera because we
+    # validate to have 1 camera only.
+    if instance.data["isolate"] and len(instance.data["setMembers"]) > 1:
+        preset["isolate"] = instance.data["setMembers"]
+
+    # Show/Hide image planes on request.
+    image_plane = instance.data.get("imagePlane", True)
+    if "viewport_options" in preset:
+        preset["viewport_options"]["imagePlane"] = image_plane
+    else:
+        preset["viewport_options"] = {"imagePlane": image_plane}
+
+    # Force viewer to False in call to capture because we have our own
+    # viewer opening call to allow a signal to trigger between
+    # playblast and viewer
+    preset["viewer"] = False
+
+    # Update preset with current panel setting
+    # if override_viewport_options is turned off
+    override_viewport_options = (
+        capture_preset["Viewport Options"]["override_viewport_options"]
+    )
+    if not override_viewport_options:
+        panel_preset = capture.parse_view(instance.data["panel"])
+        panel_preset.pop("camera")
+        preset.update(panel_preset)
+
```

Comment:
We shouldn't need to revert this ourselves since the capture module's logic itself should be doing this. If not, we should fix the bug there. That's why I'm taking this out of the context manager for `viewport_default_options`
```suggestion
    # Use Pan/Zoom from instance data instead of from preset
    preset.pop("pan_zoom", None)
    preset["camera_options"]["panZoomEnabled"] = instance.data["panZoom"]
```

Reply:
Fixed in #6081